### PR TITLE
New Electrical.Machines.Examples

### PR DESCRIPTION
**Pull request by AHaumer**
I added 2 examples (mportant for industrial applications):

- DC braking of induction machines
- Resistive braking of permanent magnet synchronous machines

### CODE REVIEW

File: Modelica/Electrical/Machines.mo [hunk 1]
```diff
@@ -2020,6 +2020,140 @@ at time tStart a load torque step is applied.</p>
...
+          annotation (Line(points={{36,10},{36,10}}, color={0,0,255}));
+        connect(plugToPin2.plug_p, terminalBox.plugSupply)
+          annotation (Line(points={{-28,40},{30,40},{30,12}}, color={0,0,255}));
+        connect(plugToPin2.pin_p, ground.p)
+          annotation (Line(points={{-32,40},{-60,40}}, color={0,0,255}));
+        annotation (experiment(
+            StopTime=25,
+            Interval=0.001,
+            Tolerance=1e-06,
+            __Dymola_Algorithm="Dassl"), Documentation(info="<html>
```
Comment (beutlich) [inline Modelica/Electrical/Machines.mo]: That solver can be savely removed.

**Comment by AHaumer** [comment]
@beutlich regarding Induction machine with DC current braking:
Try the 4 different layouts. Y2 and Y3 should run without complaint. 
D2 and D3 run in Dymola 2020 without complaint, OM 1.13.2 fails. I filed a ticket at OM.
I suspect it's a problem of index reduction, and subsequently the solver fails.
The example might be tricky but it is well defined - IMHO failing is a tool issue.

**Comment by beutlich** [comment]
Tested again in SimulationX 4.0.2: Y3 runs, but D2 and Y2 are reported to be singular (with 4 equations missing).

**Comment by AHaumer** [comment]
@beutlich Thats mysterious ...
The only things you have to do is to change settings.Layout
Both OM and Dymola don't complain about missing equatins, all 4 Layouts get compiled. 
It sems that there's difficult index reduction and choice of states. 
OM fails during simulation with D2 and D3, but Y2 and Y3 are simulated.
Dymola simulates all 4 Layouts. Seems to be a tool issue. 
All 4 Layouts are used in reality (see References in the Info layer).

**Comment by AHaumer** [comment]
I'v attached an image. All changes are done automatically according to settings.Layout changes. 
Additionally, I've split the example into 4 separated, behaviour is the same  - see enclosed.
DCbraking
[IMC_DCBraking.zip](https://github.com/modelica/ModelicaStandardLibrary/files/3390818/IMC_DCBraking.zip)

**Comment by christiankral** [comment]
Reference https://trac.openmodelica.org/OpenModelica/ticket/5574 to the OpenModelica issue.

**Comment by christiankral** [comment]
I suppose other tool vendors shall test this examples with their tools as well.

**Comment by beutlich** [comment]
> 
> 
> I suppose other tool vendors shall test this examples with their tools as well.

Forwarding. How performs Modelica.Electrical.Machines.Examples.InductionMachines.IMC_DCBraking in your tool @FanliZHOU @hubertus65 @JoeRiel ?

**Comment by AHaumer** [comment]
Maybe it's usefull to investiagte the statistics of different tools during translation, especially the choice of states and iteration variables. I'm pretty sure the model itself is ok.

**Comment by beutlich** [comment]
> 
> 
> > I suppose other tool vendors shall test this examples with their tools as well.
> 
> Forwarding. How does Modelica.Electrical.Machines.Examples.InductionMachines.IMC_DCBraking perform in your tool @FanliZHOU @henrikt-ma @hubertus65 @JoeRiel ?

All four variants simulate successfully in Wolfram SystemModeler.

**Comment by JoeRiel** [comment]
On Sat, 27 Jul 2019 14:16:25 -0700
Thomas Beutlich <notifications@github.com> wrote:

> > 
> >   
> > > I suppose other tool vendors shall test this examples with their tools as well.  
> > 
> > Forwarding. How does Modelica.Electrical.Machines.Examples.InductionMachines.IMC_DCBraking perform in your tool @FanliZHOU @henrikt-ma @hubertus65 @JoeRiel ?  
> 
> All four variants simulate successfully in Wolfram SystemModeler.
> 

Where is this new example model?  I don't see it in the standard repo.

-- 
Joe Riel

**Comment by beutlich** [comment]
> Where is this new example model?

You need to checkout the corresponding branch.

```
git checkout -b MachinesBraking master
git pull https://github.com/AHaumer/ModelicaStandardLibrary.git MachinesBraking
```

**Comment by gkurzbach** [comment]
> 
> 
> Tested again in SimulationX 4.0.2: Y3 runs, but D2 and Y2 are reported to be singular (with 4 equations missing).

I can confirm that this is a tool issue.  It will be fixed in 4.0.4.

File: Modelica/Electrical/Machines.mo [hunk 2]
```diff
@@ -2020,6 +2020,138 @@ at time tStart a load torque step is applied.</p>
...
+<h4>References</h4>
+<table border=\"0\" cellspacing=\"0\" cellpadding=\"2\">
+    <tr>
+      <td>[Fischer2017]</td>
+      <td>R. Fischer, 
+         Elektrische Maschinen, 17<sup>th</sup> ed., chapter 5.3.3.,
+        <em>Hanser</em>,
+        ISBN 978-3-446-45218-3, 2017.</td>
+    </tr>
+</html>"));
```
Comment (dietmarw) [inline Modelica/Electrical/Machines.mo]: The table is not closed causing rendering problems.

**Pull request merged**